homeassistant/components/amberelectric/coordinator.py

```diff
@@ -0,0 +1,149 @@
+    """Return true if the supplied interval is a CurrentInterval."""
+    return interval.__class__ == CurrentInterval
+
+
+def is_forecast(interval: ActualInterval | CurrentInterval | ForecastInterval) -> bool:
+    """Return true if the supplied interval is a ForecastInterval."""
+    return interval.__class__ == ForecastInterval
+
+
+def is_general(interval: ActualInterval | CurrentInterval | ForecastInterval) -> bool:
+    """Return true if the supplied interval is on the general channel."""
+    return interval.channel_type == ChannelType.GENERAL
+
+
+def is_controlled_load(
+    interval: ActualInterval | CurrentInterval | ForecastInterval,
+) -> bool:
+    """Return true if the supplied interval is on the controlled load channel."""
+    return interval.channel_type == ChannelType.CONTROLLED_LOAD
+
+
+def is_feed_in(interval: ActualInterval | CurrentInterval | ForecastInterval) -> bool:
+    """Return true if the supplied interval is on the feed in channel."""
+    return interval.channel_type == ChannelType.FEED_IN
+
+
+def first(
+    intervals: ActualInterval | CurrentInterval | ForecastInterval,
+) -> ActualInterval | CurrentInterval | ForecastInterval | None:
+    """Return the first element if it exists, otherwise return None."""
+    if len(intervals) > 0:
+        return intervals[0]
+    return None
+
+
+class AmberDataService:
+    """AmberDataService - In charge of downloading the data for a site, which all the sensors read."""
+
+    def __init__(
+        self, hass: HomeAssistant, api: amber_api.AmberApi, site_id: str
+    ) -> None:
+        """Initialise the data service."""
+        self._hass = hass
+        self._api = api
+        self._site_id = site_id
+        self.coordinator: DataUpdateCoordinator | None = None
+
+        self.data: list[ActualInterval | CurrentInterval | ForecastInterval]
+
+        self.site: Site | None = None
+        self.current_prices: dict[str, CurrentInterval | None] = {
+            ChannelType.GENERAL: None,
+            ChannelType.CONTROLLED_LOAD: None,
+            ChannelType.FEED_IN: None,
+        }
+
+        self.forecasts: dict[str, list[ForecastInterval] | None] = {
+            ChannelType.GENERAL: None,
+            ChannelType.CONTROLLED_LOAD: None,
+            ChannelType.FEED_IN: None,
+        }
+
+    @callback
+    def async_setup(self) -> None:
+        """Set up the data service."""
+        self.coordinator = DataUpdateCoordinator(
+            self._hass,
+            LOGGER,
+            name="amberelectric",
+            update_method=self.async_update_data,
+            update_interval=self.update_interval,
+        )
+
+    @property
+    def update_interval(self) -> timedelta:
+        """Update interval property."""
+        return timedelta(minutes=1)
+
+    def update(self) -> None:
+        """Update callback."""
+        try:
+            sites = list(
+                filter(lambda site: site.id == self._site_id, self._api.get_sites())
+            )
```

Comment:
More readable:
```suggestion
            sites = [site for site in self._api.get_sites() if site.id == self._site_id]
```

However, is it really necessary to fetch the sites and verify it exists on each update? how often do they change ?